web_ui/prometheus.go

```diff
@@ -470,6 +470,12 @@ func ConfigureEmbeddedPrometheus(ctx context.Context, engine *gin.Engine) error
 	}
 	cfg.tsdb.RetentionDuration = retention
 
+	maxBytes, err := units.ParseBase2Bytes(param.Monitoring_DataRetentionSize.GetString())
+	if err != nil {
+		return fmt.Errorf("failed to parse max bytes for retention size: %v", err)
+	}
+	cfg.tsdb.MaxBytes = maxBytes
```

Comment:
I tried configuring a value of `-10B` and the server started happily with no warnings or errors. Do you have any ideas what Prometheus does behind the scenes? Should handling of this number be moved into `config/config.go` so we can validate the input more rigorously?